runtime/compiler/env/VMJ9.cpp

```diff
@@ -7644,6 +7644,53 @@ TR_J9VM::inlineNativeCall(TR::Compilation * comp, TR::TreeTop * callNodeTreeTop,
          callNode = TR::Node::createWithSymRef(TR::aloadi, 1, 1, callNode, comp->getSymRefTab()->findOrCreateJavaLangClassFromClassSymbolRef());
          return callNode;
 
+      case TR::java_lang_Class_getStackClass:
+         {
+         if ((isAOT_DEPRECATED_DO_NOT_USE()
+                  && !comp->getOption(TR_UseSymbolValidationManager))
+               || !callNode->getFirstChild()->getOpCode().isLoadConst())
+            {
+            return 0;
+            }
+
+         int32_t callerIndex = callNode->getByteCodeInfo().getCallerIndex();
+         int32_t stackIndex = callNode->getFirstChild()->getInt();
+         int32_t stackIterator = 0;
+         if (stackIndex < 0)
+            return 0;
+
+         // j/l/C.getStackClass returns the class object at the stackIndex.
+         // Following loop walks stack to see if we have inlined the call chain
+         // such that we already know the J9Class* at the given constant stack
+         // index.
+         while (stackIterator != stackIndex)
+            {
+            if (callerIndex == -1)
+               break;
+            callerIndex = comp->getInlinedCallSite(callerIndex)._byteCodeInfo.getCallerIndex();
+            stackIterator++;
+            }
+
+         if (stackIterator == stackIndex)
+            {
+            J9Class *clazz = callerIndex == -1 ? reinterpret_cast<J9Class *>(comp->getJittedMethodSymbol()->getResolvedMethod()->classOfMethod()) :
+                                                 reinterpret_cast<J9Class *>(comp->getInlinedResolvedMethod(callerIndex)->containingClass());
+            int32_t classNameLength;
+            char *className = getClassNameChars(reinterpret_cast<TR_OpaqueClassBlock *>(clazz), classNameLength);
+            if (performTransformation(comp, "O^O inlineNativeCall: Optimize getStackClass call [%p] with loading Class object of of %.*s for '%s' at bytecode %d\n", callNode,
```

Comment:
Two nits on the transformation diagnostic message:

1. double `of of`.
2. Won't the 3rd %s parameter print out `java/lang/Class.getStackClass(I)` with: `callNode->getSymbolReference()->getName(comp->getDebug())`?  It's arguably redundant with `getStackClass` mentioned earlier in the message.

Reply:
Fixed in https://github.com/eclipse-openj9/openj9/compare/a15ccbf3ddc577c4abb2cc8f822a196139ce17c5..591fd57f5c3e9f572ed5c907ef64476387d3cbc8